models/_generate_layers.py

```diff
@@ -496,6 +497,46 @@ def collect_layers():
   return ls
 
 
+def get_super_call_kwargs_repr(layer: LayerBase) -> str:
+  """
+  Inspects the given layer in RETURNN, extracts the super call from there, removes unwanted parameters.
+
+  :param layer: Layer to be inspected
+  :return: A string containing all arguments of the super() call of the layer which are not excluded.
+           This then can directly be used when writing the super call for the layer for the _generated_layers file
+  """
+  # get code as string list
+  code = inspect.getsource(layer.__init__).splitlines()
+
+  # get list of lines starting with super and take the first one
+  lines = [line.strip() for line in code if line.strip().startswith("super")]
+  if not lines:
+    return ''
+
+  # reformat the super call to extract what we need
+  # get the first line which contains super to get the super call
+  super_call = lines[0]  # super_call="super(ChoiceLayer, self).__init__(beam_size=beam_size, search=search, **kwargs)"
+  # only keep the part after init of super_call
+  call = super_call.split(".")[1]  # call = "__init__(beam_size=beam_size, search=search, **kwargs)"
+  assert call.startswith("__init__")
+  assert call.endswith(")")
+  # remove "__init__(" and ") from the call string
+  call_pruned = call[len("__init__("):-len(")")]  # call pruned = "beam_size=beam_size, search=search, **kwargs"
+
+  # get list of tuples for parameter with (param_name, value)
+  tup_ls = [x.split("=") for x in call_pruned.split(",") if x.strip() != "**kwargs"]
+
+  # additionally excluded parameters
+  excl_params = ["batch_norm"]
+  # remove excluded params
+  tup_ls = [
+    f"{key.strip()}={value.strip()}"
```

Comment:
Why do you need `strip` for `key`?

Why do you need `strip` for `value`?

In general, I don't like the `strip` too much at this point anyway. Esp it is bad because you do it three times for `key`, which is repeated code.
I think I would do that separately, above:
```
tup_ls = [(key.strip(), value.strip()) for (key, value) in tup_ls]
```

Reply:
This would sometimes cause extra whitespaces in the super call, which e.g. would be filtered out by the tests. I was not sure where exactly but from what I found in both key and value and thought it wouldn't harm to just strip them here, but I agree doing that beforehand is better.